src/Stratis.Bitcoin.IntegrationTests/Wallet/SendingToAndFromManyAddressesSpecification_Steps.cs

```diff
@@ -0,0 +1,178 @@
+                , WalletPassword
+                , nodeTwoRecipients
+                , FeeType.Medium
+                , 101);
+
+            Transaction transaction = this.nodes[NodeOne].FullNode.WalletTransactionHandler().
+                BuildTransaction(this.transactionBuildContext);
+
+            transaction.Outputs.Count.Should().Be(51);
+
+            this.nodes[NodeOne].FullNode.NodeService<WalletController>().
+                SendTransaction(new SendTransactionRequest(transaction.ToHex()));
+
+            TestHelper.WaitLoop(() => this.nodes[NodeTwo].CreateRPCClient().GetRawMempool().Length > 0);
+            TestHelper.WaitLoop(() => this.nodes[NodeTwo].FullNode.WalletManager().GetSpendableTransactionsInWallet(WalletName).Any());
+        }
+
+        private void node2_receives_the_funds()
+        {
+            this.nodeTwoTotalAmount = this.nodes[NodeTwo].FullNode.WalletManager()
+                .GetSpendableTransactionsInWallet(WalletName)
+                .Sum(s => s.Transaction.Amount);
+
+            this.nodeTwoTotalAmount.Should().Be(5000000000);
+
+            this.nodes[NodeTwo].FullNode.WalletManager()
+                .GetSpendableTransactionsInWallet(WalletName)
+                .First().Transaction.BlockHeight
+                .Should().BeNull();
+
+            this.nodes[NodeTwo].FullNode.WalletManager()
+                .GetSpendableTransactionsInWallet(WalletName)
+                .Count().Should().Be(50);
+
+            this.nodes[NodeOne].GenerateStratisWithMiner(1);
+
+            this.sharedSteps.WaitForBlockStoreToSync(this.nodes[NodeTwo], this.nodes[NodeOne]);
+
+            this.nodes[NodeTwo].FullNode.WalletManager()
+                .GetSpendableTransactionsInWallet(WalletName)
+                .First().Transaction.BlockHeight
+                .Should().Be(this.CoinBaseMaturity + 51);
+        }
+
+        private void node2_sends_funds_to_node1_FROM_fifty_addresses()
+        {
+            HdAddress sendToNodeOne = this.nodes[NodeOne].FullNode.WalletManager()
+                .GetUnusedAddress(new WalletAccountReference(WalletName, WalletAccountName));
+
+            this.transactionBuildContext = new TransactionBuildContext(
+                new WalletAccountReference(WalletName, WalletAccountName),
+                new[]
+                {
+                    new Recipient
+                    {
+                        Amount = this.nodeTwoTotalAmount - Money.COIN,
+                        ScriptPubKey = sendToNodeOne.ScriptPubKey
+                    }
+                }.ToList(), WalletPassword);
+
+            Transaction transaction = this.nodes[NodeTwo].FullNode.WalletTransactionHandler()
+                        .BuildTransaction(this.transactionBuildContext);
+
+            transaction.Inputs.Count.Should().Be(50);
+
+            this.nodes[NodeTwo].FullNode.NodeService<WalletController>()
+                .SendTransaction(new SendTransactionRequest(transaction.ToHex()));
+        }
```

Comment:
you didn't do the mempool check here like you did on the other send.

Reply:
Dropped the mempool check as it's not adding anything to the test.